modules/shells/posix.nix

```diff
@@ -69,6 +69,14 @@ in
         text = ''
           ${exportedEnvVars}
 
+          # Set path to current environment root for convenient usage in hooks.
+          FLOX_ENV="${builtins.placeholder "out"}";
+
+          # If the user has an `/etc/profile' script, source it.
+          if [[ -r "$FLOX_ENV/etc/profile" ]]; then
+            . "$FLOX_ENV/etc/profile";
+          fi
```

Comment:
What's the advantage to asking a user to create an `/etc/profile` script as opposed to putting something in `config.shell.hook`?

Reply:
I can install a "shared" `/etc/profile` script and use it in many environments without copy/pasting